src/ledger.rs

```diff
@@ -1,18 +1,162 @@
 //! The `ledger` module provides functions for parallel verification of the
 //! Proof of History ledger.
 
-use bincode::{deserialize, serialize_into};
+use bincode::{deserialize, deserialize_from, serialize_into};
 use entry::Entry;
 use hash::Hash;
 use packet::{self, SharedBlob, BLOB_DATA_SIZE};
 use rayon::prelude::*;
 use result::{Error, Result};
 use std::collections::VecDeque;
-use std::io::Cursor;
+use std::fs::{create_dir_all, File, OpenOptions};
+use std::io;
+use std::io::prelude::*;
+use std::io::{Cursor, ErrorKind, Seek, SeekFrom};
+use std::mem::size_of;
+use std::path::Path;
 use transaction::Transaction;
 
-// a Block is a slice of Entries
+// ledger window
+#[derive(Debug)]
+pub struct LedgerWindow {
+    index: File,
+    data: File,
+}
+
+// use a CONST because there's a cast, and we don't want "sizeof::<u64> as u64"...
+const SIZEOF_U64: u64 = size_of::<u64>() as u64;
+
+impl LedgerWindow {
+    // opens a Ledger in directory, provides "infinite" window
+    pub fn new(directory: &str) -> io::Result<Self> {
+        let directory = Path::new(&directory);
+
+        let index = File::open(directory.join("index"))?;
+        let data = File::open(directory.join("data"))?;
+
+        Ok(LedgerWindow { index, data })
+    }
+
+    pub fn get_entry(&mut self, index: u64) -> io::Result<Entry> {
```

Comment:
Implement the Index trait instead? https://doc.rust-lang.org/std/ops/trait.Index.html

Reply:
Index takes &self, I need &mut self...